### ISSUE TITLE
using'd raw_union within struct cannot be initialized

### ISSUE THREAD

**Issue by Tetralux** [odin-lang/Odin#512]
On [current master](c43d17bfec9cc99f4d282fe05c5579e318825c2a), Windows x64:
```go
using import "core:fmt"

main :: proc() {
	S :: struct #raw_union {
		x: int,
		y: f64,
	};

	SS :: struct {
		using s: S,
		i: int,
	};

	a := SS{ .x = 42, i = 7 }; // .x should work because `using`
	println(a);
}
```
This results in the error:
```
Invalid field name '.x' in structure literal
```

**Comment by gingerBill** [comment]
As the error message suggests, this is not a bug. The issue is that assigning to nested fields can be quite difficult in practice to know exactly what it should be indexed too. 

This is not to say this isn't possible, it's just not something I have implemented yet.

**Comment by Tetralux** [comment]
Isn't this exactly the same as this?
```go
a: SS;
a.x = 42;
a.i = 7;
```

**Comment by gingerBill** [comment]
Not exactly. The difference here is the implicit indirection. In a compound literal, there is no indirection. Using a selector expression (`a.b`), there could be indirection.

It's just a technical difference.

**Comment by Tetralux** [comment]
🤔 
So you can't set "indirect" fields in a compound literal, but only because determining which nested struct the field belongs to isn't currently implemented for those literals?

**Comment by gingerBill** [comment]
Kind of.
